### PR TITLE
Add implementation of ConvexMethod members

### PR DESCRIPTION
**Pull request by hungarianhomie**

### CODE REVIEW

**Comment by hungarianhomie** [comment]
This is still not unit tested!
Some test will be added in follow on commits

File: src/ConvexMethod.cpp [hunk 1]
```diff
@@ -22,47 +52,204 @@ std::optional<LeftRightResults> ConvexMethod::classify(const geometry_msgs::msg:
...
+    if (std::fabs(area) > params.convex_hull_area_threshold) {
+        is_valid = true;
+    }
+
+    return is_valid;
 }
 
-Scenario ConvexMethod::determine_scenario() {
+Scenario ConvexMethod::determine_scenario(const std::vector<cv::Point2d>& detections_2d) {
     //TODO impl
```
Comment (hungarianhomie) [inline src/ConvexMethod.cpp:102]: Remove TODO and others missed

File: include/obj_planner/ConvexMethod.hpp [hunk 2]
```diff
@@ -15,53 +22,93 @@ enum class Scenario : std::uint8_t { kStraight = 0, kLeft = 1U, kRight = 2U };
...
+
+protected:
+    explicit TurningScenarioClassifier(const TurningScenarioParams& params) : params{params} {}
+
+    virtual std::size_t find_start_idx_on_convex_hull(const std::vector<double>& convex_hull_angles) = 0;
+    virtual std::vector<cv::Point2d>& get_outside_detections_result(LeftRightResults& classification) = 0;
+    virtual std::vector<cv::Point2d>& get_inside_detections_result(LeftRightResults& classification) = 0;
+
+private:
+    std::size_t find_end_segment_index(const std::size_t start_idx, const std::vector<cv::Point2d>& convex_hull);
```
Comment (hungarianhomie) [inline include/obj_planner/ConvexMethod.hpp]: Add some description here and other methods

File: src/ConvexMethod.cpp [hunk 3]
```diff
@@ -1,18 +1,48 @@
...
+/// Wrap angle (radians) between [0, 2pi)
+double wrap_to_2pi(double angle) {
+    angle = fmod(angle, 2.0 * pi);
+    if (angle < 0.0) angle += 2.0 * pi;
+    return angle;
+}
+
+/// Only let number reside in [0, 1]
+void clamp_between_0_and_1(double& number) {
+    if (number < 0.0) {
```
Comment (andyblarblar) [inline src/ConvexMethod.cpp:23]: Not a huge deal, but std::clamp does exist

File: src/ConvexMethod.cpp [hunk 4]
```diff
@@ -22,47 +52,204 @@ std::optional<LeftRightResults> ConvexMethod::classify(const geometry_msgs::msg:
...
 
-LeftRightResults StraightScenarioClassifier::classify(const std::vector<cv::Point>& convex_hull,
-                                                      const std::vector<cv::Point>& detections_2d) {
-    //TODO impl
-    (void)convex_hull;
-    (void)detections_2d;
-    return {};
+LeftRightResults StraightScenarioClassifier::classify(const std::vector<cv::Point2d>& convex_hull,
+                                                      const std::vector<cv::Point2d>& detections_2d) {
+    (void)convex_hull;  // Unused but needed to satisfy interface. This hurts me deeply
```
Comment (andyblarblar) [inline src/ConvexMethod.cpp]: Whats the reasoning for this? Never seen this before.

Comment (hungarianhomie) [reply src/ConvexMethod.cpp]: Thanks for making this better!

File: src/ConvexMethod.cpp (hunk 3 above)
Comment (hungarianhomie) [reply src/ConvexMethod.cpp:23]: Completely forgot about std::clamp, Ill update to use that instead

Comment (hungarianhomie) [reply src/ConvexMethod.cpp:23]: For now I didn't update this to prioritize getting it tested and working.

**Pull request merged**